### PR TITLE
[JENKINS-58165] Handle parameterized GC logs filenames

### PR DESCRIPTION
**Pull request by Dohbedoh**
[JENKINS-58165](https://issues.jenkins-ci.org/browse/JENKINS-58165): Handle parameterized GC logs files name.

### CODE REVIEW

File: src/main/java/com/cloudbees/jenkins/support/impl/GCLogs.java [hunk 1]
```diff
@@ -147,6 +147,10 @@ private String getGcLogFileLocation() {
         }
         return gcLogSwitch.substring(GCLOGS_JRE_SWITCH.length());
     }
+    
+    public boolean isFileLocationParameterized(String fileLocation) {
+        return Pattern.compile(".*%[tnp].*").matcher(fileLocation).find();
```
Comment (varyvol) [inline src/main/java/com/cloudbees/jenkins/support/impl/GCLogs.java]: Is it possible a \n in the file location?

File: src/main/java/com/cloudbees/jenkins/support/impl/GCLogs.java [hunk 2]
```diff
@@ -95,7 +95,7 @@ public boolean isSelectedByDefault() {
     /**
      * Two cases:
      * <ul>
-     * <li>The file name contains <code>%t</code> or <code>%p</code> somewhere in the middle:
+     * <li>The file name contains <code>%t</code>, <code>%p</code> or <code>%n</code> somewhere in the middle:
```
Comment (MRamonLeon) [inline src/main/java/com/cloudbees/jenkins/support/impl/GCLogs.java]: I didn't find any reference to **%n**. What does it mean? See: 

- https://docs.oracle.com/en/java/javase/11/tools/java.html#GUID-BE93ABDC-999C-4CB5-A88B-1994AAAC74D5
- https://blog.codefx.org/java/unified-logging-with-the-xlog-option/

File: src/main/java/com/cloudbees/jenkins/support/impl/GCLogs.java (hunk 1 above)
Comment (MRamonLeon) [reply src/main/java/com/cloudbees/jenkins/support/impl/GCLogs.java]: It's %n, not resolved to \n I guess, but I couldn't find it documented. Just %t and %p

Comment (Dohbedoh) [reply src/main/java/com/cloudbees/jenkins/support/impl/GCLogs.java]: `%n` was actually a feature request. But you are right it is not actually supported.

**Comment by aheritier** [comment]
@Dohbedoh let's move forward ? cc @Evildethow

File: src/main/java/com/cloudbees/jenkins/support/impl/GCLogs.java (hunk 2 above)
Comment (Dohbedoh) [reply src/main/java/com/cloudbees/jenkins/support/impl/GCLogs.java]: ditto

**Pull request merged**